modules/drv-parts.nix

```diff
@@ -39,13 +38,9 @@ in {
               # select mkDerivation as a backend for this package
               imports = [drv-parts.modules.mkDerivation];
 
-              # Define dependencies from the "outer world" only via `deps`.
-              # This allows for easy overriding later.
-              deps = {pkgs, ...} {
-                inherit (pkgs)
-                  fetchurl
-                  python
-                  ;
+              # Define dependencies from the "outer world" only via `depsFrom`.
+              # This allows for easy overriding via `config.deps` later.
+              depsFrom = {inherit (pkgs) fetchurl python};
```

Comment:
Where does this `pkgs` come from?
The point of having a function here is to avoid bringing `pkgs` into the larger scope and seducing people into using it directly.

Reply:
This example is bad. `depsFrom` should never be specified from within the package module itself. It should be set from outside.